Missing user_fpxregs_struct on musl x86

[user_fpxregs_struct][1] does exist in musl, but doesn't seem to be available in the libc crate:
```
10:34.49 error[E0432]: unresolved import `libc::user_fpxregs_struct`
10:34.49  --> /builds/alpine/aports/community/firefox/src/firefox-120.0.1/third_party/rust/minidump-writer/src/linux/thread_info/x86.rs:5:5
10:34.49   |
10:34.49 5 | use libc::user_fpxregs_struct;
10:34.49   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ no `user_fpxregs_struct` in the root
10:34.52 error[E0432]: unresolved imports `libc::user`, `libc::user_fpregs_struct`, `libc::user_regs_struct`
10:34.52  --> /builds/alpine/aports/community/firefox/src/firefox-120.0.1/third_party/rust/minidump-writer/src/linux/thread_info/x86.rs:7:12
10:34.52   |
10:34.52 7 | use libc::{user, user_fpregs_struct, user_regs_struct};
10:34.52   |            ^^^^  ^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^ no `user_regs_struct` in the root
10:34.52   |            |     |
10:34.52   |            |     no `user_fpregs_struct` in the root
10:34.52   |            no `user` in the root
10:34.55 error[E0412]: cannot find type `_libc_fpreg` in crate `libc`
10:34.55   --> /builds/alpine/aports/community/firefox/src/firefox-120.0.1/third_party/rust/minidump-writer/src/linux/crash_context/x86.rs:53:71
10:34.55    |
10:34.55 53 |             debug_assert_eq!(fs._st.len() * std::mem::size_of::<libc::_libc_fpreg>(), 80);
10:34.55    |                                                                       ^^^^^^^^^^^ not found in `libc`
10:34.55 error[E0412]: cannot find type `_libc_fpreg` in crate `libc`
10:34.55   --> /builds/alpine/aports/community/firefox/src/firefox-120.0.1/third_party/rust/minidump-writer/src/linux/crash_context/x86.rs:57:62
10:34.55    |
10:34.55 57 |                     fs._st.len() * std::mem::size_of::<libc::_libc_fpreg>(),
10:34.55    |                                                              ^^^^^^^^^^^ not found in `libc`
```

[1]: https://elixir.bootlin.com/musl/v1.2.4/source/arch/i386/bits/user.h#L8

--- a/src/unix/linux_like/linux/musl/b32/x86/mod.rs
+++ b/src/unix/linux_like/linux/musl/b32/x86/mod.rs
@@ -157,6 +157,22 @@ s! {
 }
 
 s_no_extra_traits! {
+    pub struct user_fpxregs_struct {
+        pub cwd: ::c_ushort,
+        pub swd: ::c_ushort,
+        pub twd: ::c_ushort,
+        pub fop: ::c_ushort,
+        pub fip: ::c_long,
+        pub fcs: ::c_long,
+        pub foo: ::c_long,
+        pub fos: ::c_long,
+        pub mxcsr: ::c_long,
+        __reserved: ::c_long,
+        pub st_space: [::c_long; 32],
+        pub xmm_space: [::c_long; 32],
+        padding: [::c_long; 56],
+    }
+
     pub struct ucontext_t {
         pub uc_flags: ::c_ulong,
         pub uc_link: *mut ucontext_t,
@@ -169,6 +185,64 @@ s_no_extra_traits! {
 
 cfg_if! {
     if #[cfg(feature = "extra_traits")] {
+        impl PartialEq for user_fpxregs_struct {
+            fn eq(&self, other: &user_fpxregs_struct) -> bool {
+                self.cwd == other.cwd
+                    && self.swd == other.swd
+                    && self.twd == other.twd
+                    && self.fop == other.fop
+                    && self.fip == other.fip
+                    && self.fcs == other.fcs
+                    && self.foo == other.foo
+                    && self.fos == other.fos
+                    && self.mxcsr == other.mxcsr
+                // Ignore __reserved field
+                    && self.st_space == other.st_space
+                    && self.xmm_space == other.xmm_space
+                // Ignore padding field
+            }
+        }
+
+        impl Eq for user_fpxregs_struct {}
+
+        impl ::fmt::Debug for user_fpxregs_struct {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("user_fpxregs_struct")
+                    .field("cwd", &self.cwd)
+                    .field("swd", &self.swd)
+                    .field("twd", &self.twd)
+                    .field("fop", &self.fop)
+                    .field("fip", &self.fip)
+                    .field("fcs", &self.fcs)
+                    .field("foo", &self.foo)
+                    .field("fos", &self.fos)
+                    .field("mxcsr", &self.mxcsr)
+                // Ignore __reserved field
+                    .field("st_space", &self.st_space)
+                    .field("xmm_space", &self.xmm_space)
+                // Ignore padding field
+                    .finish()
+            }
+        }
+
+        impl ::hash::Hash for user_fpxregs_struct {
+            fn hash<H: ::hash::Hasher>(&self, state: &mut H) {
+                self.cwd.hash(state);
+                self.swd.hash(state);
+                self.twd.hash(state);
+                self.fop.hash(state);
+                self.fip.hash(state);
+                self.fcs.hash(state);
+                self.foo.hash(state);
+                self.fos.hash(state);
+                self.mxcsr.hash(state);
+                // Ignore __reserved field
+                self.st_space.hash(state);
+                self.xmm_space.hash(state);
+                // Ignore padding field
+            }
+        }
+
         impl PartialEq for ucontext_t {
             fn eq(&self, other: &ucontext_t) -> bool {
                 self.uc_flags == other.uc_flags

